version: '3.8'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: vetexams-app
    ports:
      - "${APP_PORT:-8000}:80"
    environment:
      WWWGROUP: '${WWWGROUP:-1000}'
      LARAVEL_SAIL: 1
      XDEBUG_MODE: '${SAIL_XDEBUG_MODE:-off}'
      XDEBUG_CONFIG: '${SAIL_XDEBUG_CONFIG:-client_host=host.docker.internal}'
      IGNITION_LOCAL_SITES_PATH: '${PWD}'
    volumes:
      - '.:/var/www/html'
    networks:
      - vetexams
    depends_on:
      - postgres

  postgres:
    image: 'postgres:15'
    container_name: vetexams-postgres
    ports:
      - '${FORWARD_DB_PORT:-5432}:5432'
    environment:
      PGPASSWORD: '${DB_PASSWORD}'
      POSTGRES_DB: '${DB_DATABASE}'
      POSTGRES_USER: '${DB_USERNAME}'
      POSTGRES_PASSWORD: '${DB_PASSWORD}'
    volumes:
      - 'vetexams-postgres:/var/lib/postgresql/data'
      - './database/init:/docker-entrypoint-initdb.d'
    networks:
      - vetexams
    healthcheck:
      test:
        - CMD
        - pg_isready
        - '-q'
        - '-d'
        - '${DB_DATABASE}'
        - '-U'
        - '${DB_USERNAME}'
      retries: 3
      timeout: 5s

networks:
  vetexams:
    driver: bridge

volumes:
  vetexams-postgres:
    driver: local